capture_service/perfetto_trace.cc

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2024 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "perfetto_trace.h"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <perfetto.h>
+#include <chrono>
+#include <string>
+#include <thread>
+
+#include "log.h"
+
+PERFETTO_DEFINE_CATEGORIES(
+perfetto::Category("dive").SetDescription("Dive layer events"),
+perfetto::Category("gpu_renderstage").SetDescription("Rendering and graphics events"));
+
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
+
+namespace Dive
+{
+
+namespace
+{
+// TODO(renfeng): determine how long should be the trace, and pass in the value here.
+int32_t kTraceDurationMs = 5000;
+}  // namespace
+PerfettoTraceManager::PerfettoTraceManager()
+{
+    InitializePerfetto();
+}
+
+void PerfettoTraceManager::InitializePerfetto()
+{
+    // TODO: setprop debug.graphics.gpu.profiler.perfetto 1
+    LOGI("In InitializePerfetto \n");
+    perfetto::TracingInitArgs args;
+    args.backends = perfetto::kSystemBackend;
+    args.backends |= perfetto::kInProcessBackend;
+
+    perfetto::Tracing::Initialize(args);
+
+    bool registered = perfetto::TrackEvent::Register();
+    LOGI("TrackEvent Registered %d \n", registered);
+}
+
+void PerfettoTraceManager::StartNewSession(const std::string &trace_file_name)
+{
+    LOGI("Begin schedule trace session\n");
+    m_tracing_worker.Schedule([trace_file_name] {
+        LOGI("In StartTracing \n");
+        perfetto::TraceConfig cfg;
+        auto                 *buffers = cfg.add_buffers();
+        cfg.set_flush_period_ms(1000);
+        buffers->set_size_kb(131072);
```

Comment:
out of curiosity, where do you get this 131072 (128k)? is it something we might increase in the future?

Reply:
would there be a crash or memory corruption when the size is not big enough?

Reply:
so corruption could potentially happen if the size is not big enough? my worry is that if this happens silently it might be hard for us to find out, but i guess if AGI is using this value, it should be big enough.